internal/v4/stats.go

```diff
@@ -132,17 +132,35 @@ func (h *ReqHandler) serveStatsUpdate(_ http.Header, r *http.Request) (interface
 		return nil, errgo.Notef(err, "cannot unmarshal body")
 	}
 
-	rid, err := h.resolveURL(req.CharmReference)
-	if err != nil {
-		return nil, errgo.Mask(err, errgo.Is(params.ErrNotFound))
-	}
+	errors := make([]error, 0)
+	for i := range req {
+		rid, err := h.resolveURL(req[i].CharmReference)
+		if err != nil {
+			errors = append(errors, errgo.Notef(err, "cannot find entity for url: %s", req[i].CharmReference))
+			continue
+		}
 
-	logger.Infof("Increase download stats for id: %s at time: %s", rid, req.Timestamp)
+		logger.Infof("Increase download stats for id: %s at time: %s", rid, req[i].Timestamp)
 
-	if err := h.Store.IncrementDownloadCountsAtTime(rid, req.Timestamp); err != nil {
-		return nil, err
+		if err := h.Store.IncrementDownloadCountsAtTime(rid, req[i].Timestamp); err != nil {
+			errors = append(errors, err)
+			continue
+		}
 	}
 
+	if len(errors) != 0 {
+		messages := make([]string, len(errors))
+		for i, err := range errors {
+			messages[i] = err.Error()
+		}
+		encodedMessages, err := json.Marshal(messages)
```

Comment:
Rather than encoding the errors as JSON (which won't look very nice if we see the error), I guess we could use the params.Error.Info field to store an error for each entry and return that. Perhaps using the string form of the index number as the key. Alternatively, just keep the first error and the count and return a message based on that, e.g.

     cannot increment stats for trusty/wordpress: cannot resolve URL: charm not found (and 2134 more errors)

That probably fits best with what we actually want to see. We can log the other errors.